lib/src/package_analysis/lower_bound_constraint_analysis.dart

```diff
@@ -0,0 +1,630 @@
+    // an access of an object property
+    super.visitPrefixedIdentifier(node);
+
+    final parentNode = node.parent;
+    if (node.staticElement != null) {
+      element = node.staticElement!;
+    } else if (parentNode is AssignmentExpression &&
+        parentNode.writeElement != null) {
+      // this special case is needed to retrieve the PropertyAccessorElement for a setter
+      element = parentNode.writeElement!;
+    } else {
+      // we cannot statically resolve what was invoked
+      return;
+    }
+
+    if (element is! PropertyAccessorElement) {
+      // a PrefixedIdentifier does not necessarily represent a property access
+      return;
+    } else if (node.prefix.staticType is FunctionType) {
+      // do not allow the parent to be a Function
+      // (this breaks the assertion node.prefix.staticType?.element != null )
+      return;
+    }
+
+    try {
+      processIdentifier(node.identifier);
+      if (element.enclosingElement3 is ExtensionElement) {
+        parentElement = element.enclosingElement3!;
+      } else {
+        parentElement =
+            node.prefix.staticType?.element2! ?? node.prefix.staticElement!;
+      }
+      identifyDependencyName();
+    } on AnalysisException {
+      // do not continue if this invocation is unfit for analysis
+      return;
+    }
+
+    late bool constraintIssue;
+
+    if (parentElement is ClassElement) {
+      // ignore generics
+      if ((parentElement as ClassElement).typeParameters.isNotEmpty) {
+        return;
+      }
+      constraintIssue = !dependencyShape.containsPropertyWithName(
+          parentElement.name!, symbolName);
+    } else if (parentElement is ExtensionElement) {
+      constraintIssue = !dependencyShape.containsExtensionPropertyWithName(
+          parentElement.name!, symbolName);
+    } else {
+      // we may be looking at a top-level getter or setter
+      // context.warning('Subclass ${parentElement.toString()} of parentElement (getter/setter) is not supported.');
+      return;
+    }
+
+    issues[elementId] = constraintIssue
+        ? LowerBoundConstraintIssue(
+      dependencyPackageName: packageName,
+            constraint: context.dependencies[packageName]!.version,
+            currentVersion: context.findInstalledVersion(packageName),
+            lowestVersion: Version.parse(dependencyShape.version),
+            identifier: symbolName,
+            parentName: parentElement.name!,
+            kind: element.kind,
+            parentKind: parentElement.kind,
+            references: [span],
+          )
+        : null;
+  }
+
+  @override
+  void visitMethodInvocation(MethodInvocation node) {
+    // an invocation of a top-level function or a class method
+    super.visitMethodInvocation(node);
+
+    if (node.methodName.staticElement == null) {
+      // we cannot statically resolve what was invoked
+      return;
+    }
+    element = node.methodName.staticElement!;
```

Comment:
does `element.declaration` help here?
what kind of syntax does it cover?

Reply:
https://pub.dev/documentation/analyzer/latest/dart_element_element/Element/declaration.html

Reply:
OK